dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageCheckOption.java

```diff
@@ -105,45 +88,35 @@ static StaticLinkageCheckOption parseArguments(String[] arguments) throws ParseE
         "report-only-reachable",
         false,
         "To report only linkage errors reachable from entry point");
+    return options;
+  }
 
-    CommandLineParser parser = new DefaultParser();
-    List<Path> jarFilePaths = new ArrayList<>();
-
-    ImmutableList.Builder<String> mavenCoordinates = ImmutableList.builder();
-    try {
-      CommandLine commandLine = parser.parse(options, arguments);
-      if (Stream.of('b', 'a', 'j').filter(commandLine::hasOption).count() > 1) {
-        throw new IllegalArgumentException(
-            "One of BOM, Maven coordinates, or jar files can be specified");
+  static ImmutableList<Path> generateInputClasspath(CommandLine commandLine)
+      throws RepositoryException, ParseException {
+    
+    ImmutableList.Builder<Path> jarFileBuilder = ImmutableList.builder();
+    
+    Splitter commaSplitter = Splitter.on(",");
+    if (commandLine.hasOption("a")) {
+      String mavenCoordinatesOption = commandLine.getOptionValue("a");
+      for (String coord : commaSplitter.split(mavenCoordinatesOption)) {
+        jarFileBuilder.addAll(StaticLinkageChecker.coordinatesToClasspath(coord));
       }
-      Splitter commaSplitter = Splitter.on(",");
-      if (commandLine.hasOption("a")) {
-        String mavenCoordinatesOption = commandLine.getOptionValue("a");
-        mavenCoordinates.addAll(commaSplitter.split(mavenCoordinatesOption));
-      }
-      if (commandLine.hasOption("j")) {
-        String jarFiles = commandLine.getOptionValue("j");
-        List<Path> jarFilesInArguments =
-            Streams.stream(commaSplitter.split(jarFiles))
-                .map(name -> (Paths.get(name)).toAbsolutePath())
-                .collect(Collectors.toList());
-        jarFilePaths.addAll(jarFilesInArguments);
-      }
-
+      return jarFileBuilder.build();
+    } else if (commandLine.hasOption("j")) {
+      String jarFiles = commandLine.getOptionValue("j");
+      List<Path> jarFilesInArguments =
+          Streams.stream(commaSplitter.split(jarFiles))
+              .map(name -> (Paths.get(name)).toAbsolutePath())
+              .collect(Collectors.toList());
+      return ImmutableList.copyOf(jarFilesInArguments);
+    } else if (commandLine.hasOption("b")) {
       String mavenBomCoordinates = commandLine.getOptionValue("b");
-
-      boolean reportOnlyReachable = commandLine.hasOption("r");
-
-      return builder()
-          .setBom(mavenBomCoordinates)
-          .setArtifacts(mavenCoordinates.build())
-          .setJarFiles(jarFilePaths)
-          .setReportOnlyReachable(reportOnlyReachable)
-          .build();
-    } catch (ParseException ex) {
-      HelpFormatter helpFormatter = new HelpFormatter();
-      helpFormatter.printHelp("StaticLinkageChecker", options);
-      throw ex;
+      // TODO(suztomo): add logic to convert Maven BOM to list of Maven coordinates as per README.md
+      
+      return jarFileBuilder.build();
```

Comment:
Let's not bother with a builder defined outside the if/else, especially since one branch ignores it. Each mutually-exclusive branch can easily stream into an `ImmutableList`.